ulauncher/modes/extensions/extension_remote.py

```diff
@@ -31,45 +31,32 @@ class ExtensionNetworkError(Exception):
     pass
 
 
-class ExtensionRemote:
+class UrlParseResult(BaseDataClass):
+    ext_id: str
+    remote_url: str
+    download_url_template: str | None = None
+
+
+class ExtensionRemote(UrlParseResult):
+    url: str
+
     def __init__(self, url: str) -> None:
         try:
-            parsed = parse_extension_url(url)
-            self.use_git = parsed["use_git"]
-            self.url = parsed["url"]
-            self.path = parsed["path"]
-            self.protocol = parsed["protocol"]
-            self.host = parsed["host"]
-
-            if not self.use_git and self.protocol != "https":
-                msg = f"Unsupported protocol {self.protocol} for {self.url}. Only HTTPS is supported."
-                raise InvalidExtensionRecoverableError(msg)  # noqa: TRY301
+            self.url = url.strip()
+            self.update(parse_extension_url(self.url))
         except Exception as e:
             logger.warning("Invalid URL: %s (%s: %s)", url, type(e).__name__, e)
             msg = f"Invalid URL: {url}"
             raise InvalidExtensionRecoverableError(msg) from e
 
-        self.ext_id = ".".join(
-            [
-                *reversed(self.host.split(".") if self.host else []),
-                *self.path.split("/"),
-            ]
-        )
         self._dir = f"{paths.USER_EXTENSIONS}/{self.ext_id}"
         self._git_dir = f"{paths.USER_EXTENSIONS}/.git/{self.ext_id}.git"
 
-    def _get_download_url(self, commit: str) -> str:
-        if self.host == "gitlab.com":
-            repo = self.path.split("/")[1]
-            return f"{self.url}/-/archive/{commit}/{repo}-{commit}.tar.gz"
-        return f"{self.url}/archive/{commit}.tar.gz"
-
```

Comment:
Replaced by creating a "template" (string with all parts replaced, except the  commit hash, which we replace later).